data-preprocessing/video_download/nsmq_vid.py

```diff
@@ -0,0 +1,59 @@
+#This python script reads video links from a given sheet in an excel document, 
+#downloads the videos and saves the videos to a google drive folder
+#K.T Yeboah | 27-02-2023| 11:44am
+
+
+#import relevant libaries
+import pandas as pd
+from pytube import YouTube
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaFileUpload
+from pathlib import Path
+
+
+#Define doc variables
+excel_document_name = 'NSMQ.xlsx'  #document name
+excel_sheet_name = 'Sheet1'   #name of sheet in document
+videos_header_name = 'Links' #name of header of column containing video links
+
+# Read the Excel sheet and extract video links
+df = pd.read_excel(excel_document_name, excel_sheet_name)
+video_links = df[videos_header_name]
+
+
+# Download videos and save them to a local directory
+local_directory = 'videos/'
+
+for link in video_links:
+    try:
+        yt = YouTube(link)
+        video = yt.streams.filter(adaptive=True, file_extension='mp4').order_by('resolution').desc().first()
+        video.download(local_directory)
```

Comment:
would be good to add a log message here to show progress. eg. video xyz downloaded successfully. 
that way the user sees the progress as the script runs